mlrun/api/crud/model_monitoring/_model_endpoint_stores/sql_model_endpoint_store.py

```diff
@@ -0,0 +1,551 @@
+        :param connection_string: Valid connection string or a path to SQL database with model endpoints table.
+        """
+
+        super().__init__(project=project)
+        self.connection_string = connection_string
+        self.table_name = model_monitoring_constants.EventFieldType.MODEL_ENDPOINTS
+
+    def write_model_endpoint(self, endpoint: mlrun.api.schemas.ModelEndpoint):
+        """
+        Create a new endpoint record in the SQL table. This method also creates the model endpoints table within the
+        SQL database if not exist.
+
+        :param endpoint: ModelEndpoint object that will be written into the DB.
+        """
+
+        engine = db.create_engine(self.connection_string)
+
+        with engine.connect():
+            if not engine.has_table(self.table_name):
+                logger.info("Creating new model endpoints table in DB")
+                # Define schema and table for the model endpoints table as required by the SQL table structure
+                metadata = db.MetaData()
+                self._get_table(self.table_name, metadata)
+
+                # Create the table that stored in the MetaData object (if not exist)
+                metadata.create_all(engine)
+
+            # Retrieving the relevant attributes from the model endpoint object
+            endpoint_dict = self.get_params(endpoint=endpoint)
+
+            # Convert the result into a pandas Dataframe and write it into the database
+            endpoint_df = pd.DataFrame([endpoint_dict])
+            endpoint_df.to_sql(
+                self.table_name, con=engine, index=False, if_exists="append"
+            )
+
+    def update_model_endpoint(self, endpoint_id: str, attributes: typing.Dict):
+        """
+        Update a model endpoint record with a given attributes.
+
+        :param endpoint_id: The unique id of the model endpoint.
+        :param attributes: Dictionary of attributes that will be used for update the model endpoint. Note that the keys
+                           of the attributes dictionary should exist in the SQL table.
+
+        """
+
+        engine = db.create_engine(self.connection_string)
+        with engine.connect():
+            # Generate the sqlalchemy.schema.Table object that represents the model endpoints table
+            metadata = db.MetaData()
+            model_endpoints_table = db.Table(
+                self.table_name, metadata, autoload=True, autoload_with=engine
+            )
+
+            # Define and execute the query with the given attributes and the related model endpoint id
+            update_query = (
+                db.update(model_endpoints_table)
+                .values(attributes)
+                .where(
+                    model_endpoints_table.c[
+                        model_monitoring_constants.EventFieldType.ENDPOINT_ID
+                    ]
+                    == endpoint_id
+                )
+            )
+            engine.execute(update_query)
```

Comment:
if update failed because the endpoint does not exist, what error message will we receive? In the V3IO you raised an MLRun exception, I would expect every subclass of Model Endpoint to do the same